x/concentrated-liquidity/incentives.go

```diff
@@ -189,29 +189,28 @@ func (k Keeper) prepareBalancerPoolAsFullRange(ctx sdk.Context, clPoolId uint64,
 		if balancerPoolLiquidity[1].Denom != clPool.GetToken1() {
 			return 0, sdk.ZeroDec(), types.ErrInvalidBalancerPoolLiquidityError{ClPoolId: clPoolId, BalancerPoolId: canonicalBalancerPoolId, BalancerPoolLiquidity: balancerPoolLiquidity}
 		}
-	} else {
+	} else if balancerPoolLiquidity[0].Denom == clPool.GetToken1() {
 		asset0Amount = balancerPoolLiquidity[1].Amount
 		asset1Amount = balancerPoolLiquidity[0].Amount
 
 		// Ensure second denom matches (bal1 -> CL0)
 		if balancerPoolLiquidity[1].Denom != clPool.GetToken0() {
 			return 0, sdk.ZeroDec(), types.ErrInvalidBalancerPoolLiquidityError{ClPoolId: clPoolId, BalancerPoolId: canonicalBalancerPoolId, BalancerPoolLiquidity: balancerPoolLiquidity}
 		}
+	} else {
+		return 0, sdk.ZeroDec(), types.ErrInvalidBalancerPoolLiquidityError{ClPoolId: clPoolId, BalancerPoolId: canonicalBalancerPoolId, BalancerPoolLiquidity: balancerPoolLiquidity}
 	}
 
 	// Calculate the amount of liquidity the Balancer amounts qualify in the CL pool. Note that since we use the CL spot price, this is
 	// safe against prices drifting apart between the two pools (we take the lower bound on the qualifying liquidity in this case).
 	// The `sqrtPriceLowerTick` and `sqrtPriceUpperTick` fields are set to the appropriate values for a full range position.
 	qualifyingFullRangeSharesPreDiscount := math.GetLiquidityFromAmounts(clPool.GetCurrentSqrtPrice(), types.MinSqrtPrice, types.MaxSqrtPrice, asset0Amount, asset1Amount)
 
-	// Get discount ratio from governance-set discount rate. Note that the case we check for is technically impossible, but we include
-	// the check as a guardrail anyway. Specifically, we error if the discount ratio is not [0, 1].
+	// Get discount ratio from governance-set discount rate.
 	// Note that discount rate is the amount that is being discounted by (e.g. 0.05 for a 5% discount), while discount ratio is what
 	// we multiply by to apply the discount (e.g. 0.95 for a 5% discount).
+	// Concentrated Liquidity parameters provide a contract that the discount rate will be between 0 and 1.
 	balancerSharesDiscountRatio := sdk.OneDec().Sub(k.GetParams(ctx).BalancerSharesRewardDiscount)
-	if !balancerSharesDiscountRatio.GTE(sdk.ZeroDec()) || !balancerSharesDiscountRatio.LTE(sdk.OneDec()) {
-		return 0, sdk.ZeroDec(), types.InvalidDiscountRateError{DiscountRate: k.GetParams(ctx).BalancerSharesRewardDiscount}
-	}
```

Comment:
I'm kind of of the mindset that if we recognize something is not currently possible, we should note the contract that this relies on and remove superfluous checks. The reason I deleted this one specifically is:

1. I verified the parameter has checks in place to prevent it from being LT 0 and GT 1
2. If we made a PR to remove this check from the parameter in the future, someone would surely flag this

I don't mind adding it back, but it just makes me think "where is the line" to keep code succinct and to the point vs having so many secondary and third double checks that things become more confusing.

Reply:
Another reason I removed this was the mutation test hard coded one and the other to false. When attempting to add test cases to ensure this was checked, the parameter checks prevented me from setting this value in the first place.

Reply:
I think the CL code generally would benefit from far more succinctness, and clarity on where guarantees are asserted